reframe/frontend/executors/policies.py

```diff
@@ -272,6 +356,9 @@ def _failall(self, cause):
         for task in self._retired_tasks:
             task.abort(cause)
 
+        for task in self._completed_tasks:
+            task.abort(cause)
```

Comment:
You could combine the killing of the waiting, retired and completed tasks in a single loop:

```python
for task in itertools.chain(self._waiting_tasks, self._retired_tasks, self._completed_tasks):
    task.abort(cause)
```